core/DeviceDetector/DeviceDetectorFactory.php

```diff
@@ -16,33 +16,52 @@ class DeviceDetectorFactory
     protected static $deviceDetectorInstances = array();
 
     /**
-     * Returns a Singleton instance of DeviceDetector for the given user agent
+     * Returns a Singleton instance of DeviceDetector for the given user agent.
      * @param string $userAgent
-     * @param bool $useFileCache
      * @return DeviceDetector
+     * @deprecated Should get a factory via StaticContainer and call makeInstance() on it instead
      */
-    public static function getInstance($userAgent, $useFileCache = true)
+    public static function getInstance($userAgent)
+    {
+        return (new DeviceDetectorFactory())->makeInstance($userAgent);
```

Comment:
Maybe better to write $factory = new DeviceDetectorFactory(); and then on newline $factory->makeInstance()